connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1013,7 +1013,12 @@ private void reconfigureConnector(final String connName, final Callback<Void> cb
                         @Override
                         public void run() {
                             try {
-                                String reconfigUrl = RestServer.urlJoin(leaderUrl(), "/connectors/" + connName + "/tasks");
+                                String leaderUrl = leaderUrl();
+                                if (leaderUrl == null || leaderUrl.trim().length() == 0)
```

Comment:
Maybe `leaderUrl.trim().isEmpty()` instead?

Reply:
Indeed, `isEmpty()` is better.